src/TraceEvent/Parsers/KernelTraceEventParser.cs

```diff
@@ -4450,7 +4507,15 @@ public enum ThreadWaitMode
         // public int PreviousCState { get { return GetByteAt(10); } }
         // public int SpareByte { get { return GetByteAt(11); } }
 
-        public ThreadWaitReason OldThreadWaitReason { get { return (ThreadWaitReason)GetByteAt(0xc); } }
+        public ThreadWaitReason OldThreadWaitReason {
+            get {
+                int res = GetByteAt(0xc);
+                // Assert it's a valid reason 
+                // TODO: this fails all the time!!
```

Comment:
I've not seen this assert fail after enabling, so I will enable it and remove TODO